pallets/circuit/src/lib.rs

```diff
@@ -439,7 +442,13 @@ pub mod pallet {
                             &vec![],
                             None,
                         );
-                        deletion_counter += 1;
+                        if let Some(v) = deletion_counter.checked_add(1) {
+                            deletion_counter = v;
+                        }
+                        // cannot return the error
```

Comment:
comment here

Reply:
Just make sure we log and short circuit via `return`

Reply:
also ideally try do the check higher up so we don't make any storage changes and fail validation at the end

Reply:
done